pkg/block/metadata/deletionmark.go

```diff
@@ -48,6 +48,17 @@ type DeletionMark struct {
 func ReadDeletionMark(ctx context.Context, bkt objstore.BucketReader, logger log.Logger, dir string) (*DeletionMark, error) {
 	deletionMarkFile := path.Join(dir, DeletionMarkFilename)
 
+	// BucketReader.Get reports missing file as a failure (via metrics),
+	// but since most blocks don't have this marker file, it skews metrics unnecessarily.
+	// By using exists first, we avoid that.
+	exists, err := bkt.Exists(ctx, deletionMarkFile)
```

Comment:
I'm dubious about having to do another API call just to check if the object exists, when we can get the same information from `Get()`. I personally believe #2365 works better ([my take here](https://github.com/thanos-io/thanos/pull/2365#issuecomment-608260426)).

Reply:
ack